### ISSUE TITLE
Re-write script to use with Lossless Audio Checker

### ISSUE THREAD

**Issue by perfect7gentleman** [sirjaren/redoflacs#4]
http://losslessaudiochecker.com
Is it possible to re-write script to use with Lossless Audio Checker?
auCDtect is outdated and do not support pure 64-bit systems.

**Comment by sirjaren** [comment]
@perfect7gentleman, interesting!

I'll have to take a look and see what we can do here.
I've always been annoyed with `auCDtect`, because of bugs with using FLAC files directly.

Thanks for the heads up!

**Comment by perfect7gentleman** [comment]
Linux version of LAC doesn't support FLACs directly. I tried to modify existing scripts to support LAC, but as I'm not expert in scripting, I failed.

**Comment by sirjaren** [comment]
I suppose I'm used to that. With `auCDtect`, there is a bug with handling WAV over a pipe, which requires the WAV file to be a file.

I'm hoping with `LAC` (I haven't yet looked at this tool), I can convert a FLAC to WAV, and send the stream directly to `LAC`. That would simplify things a bit. :smile:

I'm going to leave this issue open until work starts on this.
Since it's been years since I've worked on `redoflacs`, I want to refactor some things and clean things up first.

**Comment by sirjaren** [comment]
Looks like Lossless Audio Checker is no more. 😦 
https://losslessaudiochecker.com/

Closing this for now - maybe there will be other tools to integrate in the future.

**Comment by eadmaster** [comment]
As alternative, i'm still using aucdtect with a temp file on linux:

https://github.com/eadmaster/dotfiles/blob/main/bin/filecheck#L154-L159

**Comment by sirjaren** [comment]
> As alternative, i'm still using aucdtect with a temp file on linux:
> 
> https://github.com/eadmaster/dotfiles/blob/main/bin/filecheck#L154-L159

Exactly. auCDtect on Linux requires a WAV file as using as using a pipe/stream is broken/buggy.

`redoflacs` creates a temporary WAV file as well:
https://github.com/sirjaren/redoflacs/blob/master/redoflacs#L1542

And removes them when finished.

**Comment by perfect7gentleman** [comment]
LAC for Linux is still can be downloaded

**Comment by sirjaren** [comment]
Where can it be found and trusted? I couldn't find any reliable Linux distribution repository.

**Comment by perfect7gentleman** [comment]
https://v1r.eu/files/audio_tools/Lossless_Audio_Checker/[Linux]%20[CLI]%202.0.5/

**Comment by perfect7gentleman** [comment]
So, can we re-open issue ?

**Comment by sirjaren** [comment]
I took a quick look, it appears I can integrate LAC fairly easily into `redoflacs`.
It helped to run `strings LAC` to view the various results returned from LAC.

Additionally, there isn't a percentage of completion displayed but `=` characters are incremented, forming a progress bar.
I can count them and do some basic arithmetic to determine percentage completed.

**Comment by sirjaren** [comment]
@perfect7gentleman

I have initial work to support LAC if you want to give it a try here:
https://github.com/sirjaren/redoflacs/tree/feat_lac_support

I have some cleanup and rework to simplify duplicate code, but would like feedback if you can provide any.
Thank you!

`redoflacs -l ~/music`
or
`redoflacs -L ~/music`

**Comment by perfect7gentleman** [comment]
It works.
But it would be great if temporay wav-file is created in \tmp directory.

**Comment by sirjaren** [comment]
Great to hear!

> It works. But it would be great if temporay wav-file is created in /tmp directory.

I like that idea.

My only concern is due to the parallelism, what if a user has, say, a 32 core system and they run the `auCDtect` or `LAC` operations?
Let's say they use all 32 cores as jobs.

That would mean, up to 32 WAV files in `/tmp` which is likely `tmpfs`.
It would then be possible to fill up `/tmp`, which is not ideal.

Thoughts?

**Comment by perfect7gentleman** [comment]
Just give a warning that the more core is used the more RAM is consumed.

**Comment by sirjaren** [comment]
Yeah, that would work - since it's not unreasonable to think that space could be exhausted as it stands today with how WAV files are generated.

I'm probably going to make this a configuration item vs defaulting to `/tmp`.
That way, users can configure where temporary WAV files go, eg: `/tmp`, `/var/tmp`, `/NAS/tmp`, etc.

Great suggestion!

**Comment by perfect7gentleman** [comment]
I've tested last version of LAC-redoflacs, it works well.

**Comment by sirjaren** [comment]
Thanks!

I'm currently still refactoring a bit to share some of the code between the auCDtect and LAC operations.

Hoping to have that ironed out soon with time, and then will implement the temporary directory configuration.

**Comment by sirjaren** [comment]
@perfect7gentleman 
I have some work in place here:
https://github.com/sirjaren/redoflacs/tree/feature_decode_to_tmp_dir

Which should allow you to choose your directory.
You'll want to re-generate your `redoflacs` configuration via: `redoflacs -o` before doing so (it's non-destructive).

The configuration has some options moved around, but there is only one new option.

I have some other things to clean-up before release, but I'm getting closer.

**Comment by sirjaren** [comment]
This has been implemented and completed with the 1.1.0 release of `redoflacs`:
https://github.com/sirjaren/redoflacs/releases/tag/1.1.0

Closing.
